woocommerce/woocommerce-android
# [Passkey] Support passkey social login

Part of https://github.com/woocommerce/woocommerce-android/issues/9906

Depends of
- https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/128
- https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2894

Please refer to pe5pgL-40m for more setup context and instructions.

# Why
We require support for the Webauthn protocol in our Woo, WordPress and Jetpack apps. This PR introduces the same integration with the FluxC `Webauthn` events and calls in order to successfully login using a Security Key and the UI handling and interaction for the 2FA steps in the WPLogin library, but now extending this support for the Sign in with Google account option.

# How
Updates both FluxC and WPLogin libraries and overrides the expected contracts for the `needs2faSocial` event and respective Tracks events.

# Capture 
https://github.com/woocommerce/woocommerce-android/assets/5920403/a4c6764d-3cf1-4753-bb17-8bf04d30447d

# How to test
- Launch the Woo app
- Login with the `Continue with Google` option Instead of logging in with user name/password.
- See that the two-factor screen presents a "Use a security key" CTA
- Select that option and select your passkey stored in the Android credentials.
- See that you are correctly authenticated.

**⚠️ THE SECURITY KEY AUTH WON'T WORK IN EMULATORS, A REAL DEVICE IS REQUIRED**
